src/map.cpp

```diff
@@ -5222,7 +5222,8 @@ void map::load(const int wx, const int wy, const int wz, const bool update_vehic
 
 void map::shift_traps( const tripoint &shift )
 {
-    const tripoint offset( shift.x * SEEX, shift.y * SEEY, shift.z );
+    // Offset needs to have sign opposite to shift direction
+    const tripoint offset( -shift.x * SEEX, -shift.y * SEEY, -shift.z );
```

Comment:
The entire `traplocs` member is only used for funnels, testing with anything but funnels wouldn't show anything.